language/cc/config.go

```diff
@@ -29,34 +30,54 @@ func (*ccLanguage) CheckFlags(fs *flag.FlagSet, c *config.Config) error
 const (
 	cc_group_directive   = "cc_group"
 	cc_group_unit_cycles = "cc_group_unit_cycles"
+	cc_indexfile         = "cc_indexfile"
 )
 
 func (c *ccLanguage) KnownDirectives() []string {
 	return []string{
 		cc_group_directive,
 		cc_group_unit_cycles,
+		cc_indexfile,
 	}
 }
 
-func (*ccLanguage) Configure(c *config.Config, rel string, f *rule.File) {
+func (c *ccLanguage) Configure(config *config.Config, rel string, f *rule.File) {
 	var conf *cppConfig
-	if parentConf, ok := c.Exts[languageName]; !ok {
+	if parentConf, ok := config.Exts[languageName]; !ok {
 		conf = newCppConfig()
 	} else {
 		conf = parentConf.(*cppConfig).clone()
 	}
-	c.Exts[languageName] = conf
+	config.Exts[languageName] = conf
 
 	if f == nil {
 		return
 	}
 
+	definesIndexFiles := false
 	for _, d := range f.Directives {
 		switch d.Key {
 		case cc_group_directive:
 			selectDirectiveChoice(&conf.groupingMode, sourceGroupingModes, d)
 		case cc_group_unit_cycles:
 			selectDirectiveChoice(&conf.groupsCycleHandlingMode, groupsCycleHandlingModes, d)
+		case cc_indexfile:
+			// New indexfiles replace inherited ones
```

Comment:
Optional: change so each `cc_indexfile` adds to the list, and the full list is inherited from the parent. To clear the list, specify `cc_indexfile` by itself with no arguments.

This is kind of a hacky way to specify a directive, but `go_proto_compilers` and now `go_search` work this way. It gives full flexibility on whether you want to override or inherit.

Reply:
Good idea, I've applied it